com.unity.shadergraph/Editor/Drawing/PreviewManager.cs

```diff
@@ -486,10 +488,29 @@ public void RenderPreviews(bool requestShaders = true)
                 CollectPreviewProperties(m_NodesPropertyChanged, perMaterialPreviewProperties);
                 m_NodesPropertyChanged.Clear();
 
-                // timed nodes change every frame, so must be drawn
+                // timed nodes are animated, so they should be updated regularly
                 // (m_TimedPreviews has been pre-propagated downstream)
                 // HOWEVER they do not need to collect properties. (the only property changing is time..)
-                m_PreviewsToDraw.UnionWith(m_TimedPreviews);
+
+                // get current window FPS, clamp to what we consider a valid range
+                // this is probably not accurate for multi-monitor.. but should be relevant to at least one of the monitors
+                double windowFPS = Screen.currentResolution.refreshRate + 1.0;  // +1 to round up, since it is an integer and rounded down
+                windowFPS = Math.Min(windowFPS, 120.0);
+                windowFPS = Math.Max(windowFPS, 30.0);
+
+                var curTime = EditorApplication.timeSinceStartup;
+                var deltaTime = curTime - m_LastTimedUpdateTime;
+                bool isFocusedWindow = (EditorWindow.focusedWindow == editorWindow);
+
+                // we throttle the update rate, based on whether the window is focused and if unity is active
+                double k_NonFocusedFPS = 10.0;
+                double k_InactiveFPS = 2.0;
```

Comment:
Why not make these actual constants? Either here or up higher in the file?

Reply:
Made them into constants.